source/common/config/utility.cc

```diff
@@ -228,5 +228,11 @@ Grpc::AsyncClientFactoryPtr Utility::factoryForGrpcApiConfigSource(
   return async_client_manager.factoryForGrpcService(grpc_service, scope, false);
 }
 
+std::string Utility::normalizePath(absl::string_view path) {
+  std::vector<absl::string_view> segments = StringUtil::splitToken(path, "/", false);
+  bool has_trailing_slash = path.back() == '/';
+  return absl::StrCat("/", absl::StrJoin(segments, "/"), has_trailing_slash ? "/" : "");
```

Comment:
Q: I feel like this is a pretty heavy hammer from a perf perspective for solving a specific issue (duplicate '/' after rewrite). Could we detect the duplicate '/' and copy strings in a way in which this would perform better? cc @jmarantz who likes these types of things. :)

Reply:
Interesting. Probably most of the time the use-case is to have duplicates '/' at the beginning of the path.

Reply:
actually I have functional concerns about this PR first.